src/presence.js

```diff
@@ -0,0 +1,35 @@
+import * as R from 'ramda';
+
+/**
+ * Check if value is empty or undefined then returns null, otherwise, returns itself.
+ *
+ * @func presence
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.29.0|v2.29.0}
+ * @category Object
+ * @sig k -> null | k
+ * @param {k} val The value to test
+ * @return {null | k}
+ * @example
+ *
+ * presence({ foo: 'foo' }) // { foo: 'foo' }
+ * presence({}) // null
+ * presence(false) // null
+ * presence(true) // true
+ * presence('') // null
+ * presence('foo') // foo
+ * presence([]) // null
+ * presence([1, 2, 3]) // [1, 2, 3]
+ * presence(undefined) // null
+ * presence(0) // 0
+ * presence(null) // null
+ */
+
+const presence = R.cond([
```

Comment:
Right, the implementation seems almost exactly precise. There is one small thing that we missed here; any string containing only white-space characters should fail the `present` test.

```js
RA.presence(''); // => null
RA.presence('   '); // => null
RA.presence('\t\n'); // => null
```

This regexp should solve this case: `/^\s*$/g`

More information about expected behavior in this article: https://blog.appsignal.com/2018/09/11/differences-between-nil-empty-blank-and-present.html

Reply:
so @char0n i appreciate your suggestion, what do u think if we create first a method  `isBlank` to ramda-adjunct , so in the logic we could use it and can be used to other functionalities, what do u think about it?

Reply:
@char0n i created this PR if u guess is necessary exists an `isBlank` let me know => https://github.com/char0n/ramda-adjunct/pull/2141/files

Reply:
> so @char0n i appreciate your suggestion, what do u think if we create first a method `isBlank` to ramda-adjunct , so in the logic we could use it and can be used to other functionalities, what do u think about it?

Why not. Usual Modus operandi is that you put the helper function like in this case `isBlank` into `src/internal` and after the original PR dealing with `presence` is finished, you're return back to `src/internal/isBlank` and expose it as additional feature.

Reply:
> @char0n i created this PR if u guess is necessary exists an `isBlank` let me know => https://github.com/char0n/ramda-adjunct/pull/2141/files

Right, I'll get into reviewing the https://github.com/char0n/ramda-adjunct/pull/2141/files right away. This PR is blocked until isBlank will hit the master branch.

Reply:
@char0n  appreciate that .. thanks

Reply:
Added following commit related to `isBlank` - https://github.com/char0n/ramda-adjunct/commit/1810924bd6286af5a04e8966e80095dc87799007

I found out implementation of `isBlank` was not aligned with the  [the article](https://blog.appsignal.com/2018/09/11/differences-between-nil-empty-blank-and-present.html) we used as the reference.  `isBlank` should also return `true` for `null` value.